crates/uv-resolver/src/redirect.rs

```diff
@@ -48,7 +48,7 @@ pub(crate) fn url_to_precise(url: VerbatimParsedUrl) -> VerbatimParsedUrl {
         return url;
     };
 
-    // TODO(konsti): Remove once we carry more context on the `Dist`s.
+    // TODO(konsti): Remove once we carry more context on the `Dist` (e.g., `BranchOrTag` vs. `Tag`).
```

Comment:
@konstin - This actually looks like an import TODO.

Reply:
Where we losing this info?

Reply:
The bottleneck is `GitSourceUrl` -> `resolve_precise`